### PR TITLE
Validate Meta URL

### PR DESCRIPTION
**Pull request by Scrumplex**
- Add a QValidator for the meta URL field in settings
- Reset invalid meta URL setting

### CODE REVIEW

**Comment by TheKodeToad** [comment]
The validator seems to cause confusion (#1323).

Maybe it's best to remove it here and on the other field!

**Comment by Scrumplex** [comment]
Yeah Qt validators aren't really ideal. The other issue is about the curseforge API key though.

**Comment by TheKodeToad** [comment]
Sorry, I had a major brain fart

File: launcher/Application.cpp [hunk 1]
```diff
@@ -687,8 +687,16 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
                 m_settings->reset("PastebinCustomAPIBase");
             }
         }
-        // meta URL
-        m_settings->registerSetting("MetaURLOverride", "");
+        {
+            // Meta URL
+            m_settings->registerSetting("MetaURLOverride", "");
+
+            QUrl metaUrl = m_settings->get("MetaURLOverride").toString();
```
Comment (TheKodeToad) [inline launcher/Application.cpp]:
```suggestion
            QUrl metaUrl(m_settings->get("MetaURLOverride").toString());
```

this is both more explicit and makes more sense

**Review by TheKodeToad — APPROVED** [review]

**Review by getchoo — APPROVED** [review]

**Pull request merged**